fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainerRegistryImpl.java

```diff
@@ -45,7 +45,7 @@ public class IndexMaintainerRegistryImpl implements IndexMaintainerRegistry {
     protected static final IndexMaintainerRegistryImpl INSTANCE = new IndexMaintainerRegistryImpl();
 
     @Nonnull
-    private final Map<String, IndexMaintainerFactory> registry;
+    private Map<String, IndexMaintainerFactory> registry;
```

Comment:
Since this is not `final` anymore, the constructor call does not guarantee visibility in other threads anymore.

https://stackoverflow.com/questions/33702625/java-memory-visibility-in-constructors

Reply:
I want to avoid having to do something like this in my test code:
```
    static class ResettableIndexMaintainerRegistryImpl {
        public static void reset() {
            // warning - reflection - reset the static registry everytime we reset Guice
            try {
                IndexMaintainerRegistry registry = IndexMaintainerRegistryImpl.instance();
                Field privateField = registry.getClass().getDeclaredField("registry");
                privateField.setAccessible(true);
                privateField.set(registry, initRegistry());
            } catch (Exception e) {
                //
            }
        }

        protected static Map<String, IndexMaintainerFactory> initRegistry() {
            final Map<String, IndexMaintainerFactory> registry = new HashMap<>();
            for (IndexMaintainerFactory factory : ServiceLoaderProvider.load(IndexMaintainerFactory.class)) {
                for (String type : factory.getIndexTypes()) {
                    registry.put(type, factory);
                }
            }
            return registry;
        }
    }
```